datalake/src/main/java/com/sequenceiq/datalake/service/sdx/flowwait/task/SdxWaitTaskFactory.java

```diff
@@ -0,0 +1,20 @@
+package com.sequenceiq.datalake.service.sdx.flowwait.task;
+
+import javax.inject.Inject;
+
+import org.springframework.context.ApplicationContext;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.datalake.entity.SdxCluster;
+import com.sequenceiq.datalake.service.sdx.PollingConfig;
+
+@Service
+public class SdxWaitTaskFactory {
+
+    @Inject
+    private ApplicationContext applicationContext;
+
+    public SdxWaitTaskForCloudbreakFlow createCloudbreakFlowWaitTask(SdxCluster sdxCluster, PollingConfig pollingConfig, String pollingMessage) {
+        return applicationContext.getBean(SdxWaitTaskForCloudbreakFlow.class, sdxCluster, pollingConfig, pollingMessage);
```

Comment:
I think it would be better to use `new` here

Reply:
SdxWaitTaskForCloudbreakFlow has an injected field: if using new that would not be injected.